backend/routes/post.js

```diff
@@ -71,13 +73,18 @@ router.post('/:_pid', (request, response, next) => {
   /*  Comment / Payload / Body Structure
     {author,details,datetime}
   */
-  let currentPostId = request.params._pid
-  controllerPost
-    .addComment(currentPostId, request.body)
-    .then(data => {
-      response.send(data)
-    })
-    .catch(err => response.status(400).send(err))
+  var comment = new Comment(request.body);
+  comment.post = request.post;
+  comment.save(function(err, comment) {
+    if (err) return next(err);;
+    request.post.comments.push(comment);
```

Comment:
Its bad practice to directly mutate the parameter being passed in. in this case `comments` because it can cause unintended side effects later on if code else where depends on the comments parameter